LibOS/shim/src/fs/pipe/fs.c

```diff
@@ -40,28 +40,22 @@
 #include <linux/stat.h>
 
 static ssize_t pipe_read(struct shim_handle* hdl, void* buf, size_t count) {
-    if (!count)
-        return 0;
-
     PAL_NUM bytes = DkStreamRead(hdl->pal_handle, 0, count, buf, NULL, 0);
 
     if (!bytes)
         return -PAL_ERRNO;
 
-    assert((ssize_t)bytes > 0);
+    assert(count == 0 || (ssize_t)bytes > 0);
```

Comment:
If we are removing !count check, before calling DkStreamRead, then we end up adding it part of assert. This makes an assumption that host kernel would have returned error(negative), for count == 0. I think we should either leave this, as-it-exists, (OR) think if we really need any assert here, why not let the user-application handle the error.